src/tests/sys_tests.rs

```diff
@@ -1,105 +1,136 @@
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+        let cage = interface::cagetable_getref(1);
+        // Spawn a new child object using the fork syscall
+        cage.fork_syscall(2);
         // Search for the new cage object with cage_id = 2
-        let child_cage = interface::cagetable_getref(2); 
+        let child_cage = interface::cagetable_getref(2);
         // Assert the parent value is the the id of the first cage object
-        assert_eq!(child_cage.getpid_syscall(),1);
-        // Assert that the cage id of the child is the value passed in the original fork syscall
-        assert_eq!(child_cage.getuid(),2);
-        // Assert that the cwd is the same as the parent cage object
-        assert_eq!(child_cage.cwd.read(),cage.cwd.read())
+        assert_eq!(child_cage.getppid_syscall(), 1);
+        // Assert that the cage id of the child is the value passed in the original fork
+        // syscall
+        assert_eq!(child_cage.getuid_syscall(), -1);
+        assert_eq!(child_cage.getuid_syscall(), DEFAULT_UID as i32);
+        lindrustfinalize();
     }
 
+    #[test]
     pub fn ut_lind_exit() {
-        // Since exit function is heavily used and tested in other syscalls and their tests 
-        // We only perform preliminary checks for checking the sanity of this syscall
-        // We don't check for cases such as exiting a cage twice - since the exiting process 
-        // is handled by the NaCl runtime - and it ensures that a cage does not exit twice
-        lindrustinit(0);
+        // Since exit function is heavily used and tested in other syscalls and their
+        // tests We only perform preliminary checks for checking the sanity of
+        // this syscall We don't check for cases such as exiting a cage twice -
+        // since the exiting process is handled by the NaCl runtime - and it
+        // ensures that a cage does not exit twice acquiring a lock on TESTMUTEX
+        // prevents other tests from running concurrently, and also performs
+        // clean env setup
+        let _thelock = setup::lock_and_init();
         let cage = interface::cagetable_getref(1);
         // Call the exit call
-        assert_eq(cage.exit_syscall(EXIT_SUCCESS),EXIT_SUCCESS); 
-        lindrustfinalize(); 
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
     }
-} 
 
+    #[test]
+    pub fn ut_lind_exec() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+        let cage1 = interface::cagetable_getref(1);
+        // Spawn a new child
+        cage1.fork_syscall(2);
+        let cage2 = interface::cagetable_getref(2);
+        // Spawn exec and check if it returns 0
+        assert_eq!(cage2.exec_syscall(2), 0);
```

Comment:
I'm still a little confused on my end. Right now I'm understanding that you initialize a cage, cage1, then you call fork to "copy" everything from cage1 into the second index of the cagetable. This lets you initialize cage2, a copy of cage1. What exactly is exec doing here?

Reply:
fork creates a new child cage - copies the existing tables and metadata and creates a new child process with pid = 1 . exec then replaces the current running image with the new cage2 object

Reply:
What happens in cage2?

Reply:
so cage2 is getting replaced with cage2 again? shouldnt we try to replace cage1 with cage2?

Reply:
Oh yes - corrected that !